src/Workspaces/CoreTest/SymbolKeyTests.cs

```diff
@@ -647,6 +647,74 @@ object LocalFunction<T>()
             Assert.True(tested);
         }
 
+        [Fact, WorkItem(17702, "https://github.com/dotnet/roslyn/issues/17702")]
+        public void TestTupleWithLocalTypeReferences1()
+        {
+            var source = @"
+using System.Linq;
+
+class C
+{
+    void Method((C, int) t)
+    {
+    }
+}";
+            // Tuples store locations along with them.  But we can only recover those locations
```

Comment:
Given we also use SymbolKey to map _across_ compilations (something I find terribly foul, but...), how does this constraint impact that?